libs/dal/src/repositories/integration/integration.schema.ts

```diff
@@ -37,6 +37,9 @@ const integrationSchema = new Schema<IntegrationDBModel>(
       clientId: Schema.Types.String,
       projectName: Schema.Types.String,
       serviceAccount: Schema.Types.String,
+      requireTls: Schema.Types.String,
+      ignoreTls: Schema.Types.String,
+      tlsOptions: Schema.Types.String,
```

Comment:
Unfortunately we shouldn't do this. `tlsOptions` is a quite complicated object (https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options) so for a _quick_ implementation we would need for now to use `Schema.Types.Mixed` in the Schema.

Then also we have the problem that Web app is capturing strings and `requireTls` and `ignoreTls` should be stored as booleans (besides what's mentioned for `tlsOptions`).
That implies proper mapping in the appropriate DTOs of API and validation.

So unfortunately we can't do a simple string 2 string mapping throughout all the system. 😞

Reply:
Thanks for the review. Sure, will make the changes accordingly.

A quick question though, why are we maintaining the schema type as String even though it's boolean? I have seen the `secure` key is also capturing it as a string in web app, but DTO is defined as a boolean, and in the schema it's defined as a String. So followed the same approach with `requireTls` and `ignoreTls`

 Should we not maintain consistency overall as web app, DTO and Schema ?

Reply:
I personally think with `secure` the wrong decision was taken for a quick development. 🙁 Besides that the content is not being validated, so even if with TS we have a different type, at runtime is passing strings.
And right now we shouldn't change it without doing a migration because it would break any integration users had already set up so we can only do the proper thing with the new additions.

@scopsy @LetItRock tagging you to align as we would need to change some of the frontend and overall we have a misalignment of existing properties with their proper type DB wise.

Reply:
@p-fernandez Made the schema changes.

Also added a checkbox option on the web app for the fields `requireTls` and `ignoreTls`.  These values will be captured as a boolean in web app. 
Didn't make any changes to `secure` as it would break existing things.
 
Let me know if this works out.  

<img width="1319" alt="Screenshot 2023-03-22 at 4 35 46 PM" src="https://user-images.githubusercontent.com/22212945/226885350-43fba319-c22d-4d4f-aefe-55f4637ee59d.png">